integration_test/warehouse/integration_test.go

```diff
@@ -0,0 +1,3462 @@
+	"github.com/rudderlabs/rudder-server/enterprise/reporting"
+	mocksApp "github.com/rudderlabs/rudder-server/mocks/app"
+	"github.com/rudderlabs/rudder-server/warehouse"
+)
+
+const (
+	succeeded    = "succeeded"
+	waiting      = "waiting"
+	aborted      = "aborted"
+	exportedData = "exported_data"
+)
+
+func TestUploads(t *testing.T) {
+	admin.Init()
+	validations.Init()
+
+	t.Run("tracks loading", func(t *testing.T) {
+		const (
+			workspaceID           = "test_workspace_id"
+			sourceID              = "test_source_id"
+			destinationID         = "test_destination_id"
+			destinationRevisionID = "test_destination_id"
+			namespace             = "test_namespace"
+		)
+
+		pool, err := dockertest.NewPool("")
+		require.NoError(t, err)
+
+		pgResource, err := resource.SetupPostgres(pool, t)
+		require.NoError(t, err)
+		minioResource, err := resource.SetupMinio(pool, t)
+		require.NoError(t, err)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		webPort, err := kithelper.GetFreePort()
+		require.NoError(t, err)
+
+		backendConfigCh := make(chan pubsub.DataEvent)
+
+		g, gCtx := errgroup.WithContext(ctx)
+		g.Go(func() error {
+			defer close(backendConfigCh)
+
+			for {
+				select {
+				case <-gCtx.Done():
+					return nil
+				case <-time.After(time.Second):
+					backendConfigCh <- pubsub.DataEvent{
+						Data: map[string]backendconfig.ConfigT{
+							workspaceID: {
+								WorkspaceID: workspaceID,
+								Sources: []backendconfig.SourceT{
+									{
+										ID:      sourceID,
+										Enabled: true,
+										Destinations: []backendconfig.DestinationT{
+											{
+												ID:      destinationID,
+												Enabled: true,
+												DestinationDefinition: backendconfig.DestinationDefinitionT{
+													Name: whutils.POSTGRES,
+												},
+												Config: map[string]interface{}{
+													"host":             pgResource.Host,
+													"database":         pgResource.Database,
+													"user":             pgResource.User,
+													"password":         pgResource.Password,
+													"port":             pgResource.Port,
+													"sslMode":          "disable",
+													"namespace":        namespace,
+													"bucketProvider":   whutils.MINIO,
+													"bucketName":       minioResource.BucketName,
+													"accessKeyID":      minioResource.AccessKeyID,
+													"secretAccessKey":  minioResource.AccessKeySecret,
+													"useSSL":           false,
+													"endPoint":         minioResource.Endpoint,
+													"syncFrequency":    "0",
+													"useRudderStorage": false,
+												},
+												RevisionID: destinationRevisionID,
+											},
+										},
+									},
+								},
+							},
+						},
+						Topic: string(backendconfig.TopicBackendConfig),
+					}
+				}
+			}
+		})
+		g.Go(func() error {
+			return runWarehouseServer(t, gCtx, webPort, pgResource, backendConfigCh)
```

Comment:
Yes, was using errgroup because of the `backenconfig`. Was planning to remove it based on [this](https://github.com/rudderlabs/rudder-server/pull/4058#discussion_r1384676221) comment.